src/typing.py

```diff
@@ -1568,7 +1546,7 @@ def no_type_check(arg):
     if isinstance(arg, type):
         arg_attrs = arg.__dict__.copy()
         for attr, val in arg.__dict__.items():
-            if val in arg.__bases__:
+            if val in arg.__bases__ + (arg,):
```

Comment:
Why is this added `(arg,)` needed? And why not in the PY2 version? (If there's a good reason please add a comment.)